electricitymap/contrib/capacity_parsers/DE.py

```diff
@@ -0,0 +1,93 @@
+from datetime import datetime
+from logging import INFO, basicConfig, getLogger
+from typing import Any
+
+from requests import Response, Session
+
+from electricitymap.contrib.config import ZoneKey
+
+logger = getLogger(__name__)
+basicConfig(level=INFO)
+
+MODE_MAPPING = {
+    "Nuclear": "nuclear",
+    "Fossil brown coal / lignite": "coal",
+    "Fossil hard coal": "coal",
+    "Fossil gas": "gas",
+    "Fossil oil": "oil",
+    "Other, non-renewable": "unknown",
+    "Hydro Run-of-River": "hydro",
+    "Hydro pumped storage": "hydro storage",
+    "Battery Storage (Power)": "battery storage",
+    "Biomass": "biomass",
+    "Wind onshore": "wind",
+    "Wind offshore": "wind",
+    "Solar": "solar",
+}
+
+SOURCE = "Fraunhofer ISE"
+
+REQUEST_URL = "https://api.energy-charts.info/installed_power?country=de&time_step=yearly&installation_decommission=false"
+CONVERT_GW_TO_MW = 1000
+
+
+def convert_to_mw(value: float | None) -> float:
+    try:
+        return float(value) * CONVERT_GW_TO_MW
+    except:
+        return None
+
+
+def fetch_production_capacity(
+    zone_key: ZoneKey, target_datetime: datetime, session: Session
+) -> dict[str, Any] | None:
+    r: Response = session.get(REQUEST_URL)
```

Comment:
If this is run via the test parser it will crash from the lack of a logger object and that session is none.

(In case someone wants to run it without updating the config file.)

Reply:
Ah okay!

That makes sense I think we can just add an early return in the test_parser that says it doesn't support capacity.

But we can do that in another PR when we have the time.

Reply:
i will make a note of that ;)